daemon/cluster/cluster.go

```diff
@@ -446,11 +446,24 @@ func (c *Cluster) UnlockSwarm(req types.UnlockRequest) error {
 
 	c.mu.RLock()
 	state := c.currentNodeState()
-	nr := c.nr
-	c.mu.RUnlock()
-	if nr == nil || errors.Cause(state.err) != errSwarmLocked {
+
+	if !state.IsActiveManager() {
+		// when manager is not active,
+		// unless it is locked, otherwise return error.
+		if err := c.errNoManager(state); err != errSwarmLocked {
+			c.mu.RUnlock()
+			return err
+		}
+	} else {
+		// when manager is active, return an error of "not locked"
+		c.mu.RUnlock()
 		return errors.New("swarm is not locked")
 	}
```

Comment:
Thanks for your feedback. @aaronlehmann 
I think your implementation is good as it makes cod simpler and more readable. I like it as well. While I wish we could keep the current style. The reason is that I hope to keep the manager role judgement first in each place this cluster.go. When we read code, we have an initial process logic: first, judge manager role, second, do the following things. To keep consistent for logic, I think using
```
if !state.IsActiveManager() {
```
makes people feel readable and it is good, too.